backend/src/main/java/com/codesquad/sidedish/auth/AuthService.java

```diff
@@ -0,0 +1,25 @@
+package com.codesquad.sidedish.auth;
+
+import com.codesquad.sidedish.order.UserRepository;
+import com.codesquad.sidedish.order.domain.User;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthService {
+
+    private final UserRepository userRepository;
+
+    public User upsertUser(User user) {
+        Optional<User> optionalUser = userRepository.findByGithubId(user.getGithubId());
+
+        if (optionalUser.isPresent()) {
+            User findUser = optionalUser.get();
+            user = findUser.update(user);
+        }
+        userRepository.save(user);
+        return user;
```

Comment:
```suggestion
		final User savedUser = userRepository.findByGithubId(user.getGithubId())
				.map(findUser -> findUser.update(user))
				.orElse(user);

		return userRepository.save(savedUser);
```

`nit` 
이왕 optional을 사용한다면 저는 이런 코딩 형태를 선호합니다.
그것이 아니더라도 if 조건문의 `isPresent`를 `isEmpty`로 변경하면 얼리 리턴이 가능할 것 같아요.